pkg/xds/generator/egress/internal_services_generator.go

```diff
@@ -26,29 +26,28 @@ func (g *InternalServicesGenerator) Generate(
 	meshResources *core_xds.MeshResources,
 ) (*core_xds.ResourceSet, error) {
 	resources := core_xds.NewResourceSet()
-	zone := xdsCtx.ControlPlane.Zone
-	apiVersion := proxy.APIVersion
-	endpointMap := meshResources.EndpointMap
-	destinations := buildDestinations(meshResources.TrafficRoutes)
-	services := g.buildServices(endpointMap, meshResources.Mesh.ZoneEgressEnabled(), zone)
 	meshName := meshResources.Mesh.GetMeta().GetName()
 
-	g.addFilterChains(
-		apiVersion,
-		destinations,
-		proxy,
-		listenerBuilder,
-		meshResources,
-		services,
+	servicesMap := g.buildServices(meshResources.EndpointMap, meshResources.Mesh.ZoneEgressEnabled(), xdsCtx.ControlPlane.Zone)
+	services := maps.Keys(servicesMap)
+	sort.Strings(services)
```

Comment:
You can probably use https://github.com/kumahq/kuma/blob/master/pkg/util/maps/sorted_keys.go and adjust SortedKeys method to be generic

Reply:
Not sure if it's worth having this util to save one line, but I added it.